ESP8266/ESP8266.cpp

```diff
@@ -404,26 +406,36 @@ nsapi_error_t ESP8266::open_udp(int id, const char* addr, int port, int local_po
 
     _smutex.lock();
     if(local_port) {
-        done = _parser.send("AT+CIPSTART=%d,\"%s\",\"%s\",%d,%d", id, type, addr, port, local_port)
-                && _parser.recv("OK\n");
+        done = _parser.send("AT+CIPSTART=%d,\"%s\",\"%s\",%d,%d", id, type, addr, port, local_port);
     } else {
-        done = _parser.send("AT+CIPSTART=%d,\"%s\",\"%s\",%d", id, type, addr, port)
-               && _parser.recv("OK\n");
+        done = _parser.send("AT+CIPSTART=%d,\"%s\",\"%s\",%d", id, type, addr, port);
     }
 
     if (done) {
+        if (!_parser.recv("OK\n")) {
+            if (_sock_already) {
+                _sock_already = false; // To be raised again by OOB msg
+                done = close(id);
+                if (!done) { // Shouldn't happen
+                    goto EXIT;
```

Comment:
Should not happend --> Should not be tolerated.
Maybe `MBED_ERROR(...)` that causes verbal error message and stops the device.